src/Microsoft.DocAsCode.Build.Engine/DocumentBuilder.cs

```diff
@@ -427,45 +431,42 @@ private static void BuildArticle(IDocumentProcessor processor, HostService hostS
                 maxParallelism);
         }
 
-        private static void Postbuild(IDocumentProcessor processor, HostService hostService)
+        private static void Postbuild(HostService hostService)
         {
             RunBuildSteps(
-                processor.BuildSteps,
+                hostService.Processor.BuildSteps,
                 buildStep =>
                 {
-                    Logger.LogVerbose($"Processor {processor.Name}, step {buildStep.Name}: Postprocessing...");
+                    Logger.LogVerbose($"Processor {hostService.Processor.Name}, step {buildStep.Name}: Postprocessing...");
                     using (new LoggerPhaseScope(buildStep.Name))
                     {
                         buildStep.Postbuild(hostService.Models, hostService);
                     }
                 });
         }
 
-        private IEnumerable<ManifestItemWithContext> ExportManifest(InnerBuildContext buildContext, DocumentBuildContext context)
+        private IEnumerable<ManifestItemWithContext> ExportManifest(HostService hostService, DocumentBuildContext context)
         {
-            var hostService = buildContext.HostService;
-            var processor = buildContext.Processor;
-            var templateProcessor = buildContext.TemplateProcessor;
```

Comment:
Thanks, i get it now. maybe later if we need to add more context related info, a HostServiceContext is still needed for better management

---
In reply to: [66018261](https://github.com/dotnet/docfx/pull/363#discussion_r66018261) [](ancestors = 66018261)